sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/ReactiveCosmosTemplate.java

```diff
@@ -325,6 +326,17 @@ public <T> Mono<T> findById(Object id, Class<T> domainType) {
     public <T> Mono<T> findById(String containerName, Object id, Class<T> domainType) {
         Assert.hasText(containerName, "containerName should not be null, empty or only whitespaces");
         Assert.notNull(domainType, "domainType should not be null");
+        CosmosEntityInformation<?, ?> cosmosEntityInformation = CosmosEntityInformation.getInstance(domainType);
+        String containerPartitionKey = cosmosEntityInformation.getPartitionKeyFieldName();
+        if ("id".equals(containerPartitionKey) && id != null) {
+            try {
+                return findById(id, domainType, new PartitionKey(id));
+            } catch (CosmosAccessException e) {
+                LOGGER.warn(e.getMessage());
+                return null;
```

Comment:
This cannot return null. The API expects a Mono<T> - a reactive type. We cannot return null here. 
Also, catching the exception is incorrect here.